File: Second Semester(Java and C++)/List 3/Java/Task 1/Quadrangle.java
import java.util.ArrayList;

public abstract class Quadrangle extends Shape {
    protected ArrayList<Integer> sides = new ArrayList<Integer>();
    protected int angle;

    @Override
    public double getArea() {
        return ((double)sides.get(0)) * ((double)sides.get(2)) * Math.sin(Math.toRadians(angle));
    }

    @Override
    public double getPerimeter() {
        double result = 0.0;
        for (int side : sides) result += (double)side;
        return result;
    }
}

File: Second Semester(Java and C++)/List 3/C++/Quadrangle.cpp
#include "Quadrangle.hpp"

#include <cmath>

double Quadrangle::getArea() noexcept(true) {
    return sides[0] * sides[2] * sin(angle * (M_PI / 180.0));
}

double Quadrangle::getPerimeter() noexcept(true) {
    double result = 0.0;
    for (int i = 0; i < 4; i++) result += sides[i];
    return result; 
}